### PR TITLE
Gardening insert at on rails 6 plus

### PR DESCRIPTION
**Pull request by altintx**
Rails 6 compatability
Upgrade bundled rails_app to 6
upgrade circle CI to current image versions
Bump ruby version to 2.7 for happier newer Rails. Still runs on Ruby 2.5.

### CODE REVIEW

File: priv/migrations/add_railway_ipc_consumed_messages.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # frozen_string_literal: true
 
-class AddRailwayIpcConsumedMessages < ActiveRecord::Migration
+class AddRailwayIpcConsumedMessages < ActiveRecord::Migration[4.2]
```
Comment (lorgio-jimenez) [inline priv/migrations/add_railway_ipc_consumed_messages.rb:3]: Would this cause issues from rails 5 to rails 6?

File: spec/support/rails_app/Gemfile [hunk 2]
```diff
@@ -25,4 +25,4 @@ end
 gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
 
 gem 'railway-ipc', path: '../../..'
-gem 'google-protobuf', '~> 3.9'
+gem 'google-protobuf', '~> 3.20.0'
```
Comment (lorgio-jimenez) [inline spec/support/rails_app/Gemfile:28]: This is a pretty big jump in the protobuf library, albeit they're minor versions, and semantic versioning usually states that minor versions include new functionality,  but were there any degradation messages?

File: priv/migrations/add_railway_ipc_consumed_messages.rb (hunk 1 above)
Comment (altintx) [reply priv/migrations/add_railway_ipc_consumed_messages.rb:3]: It's mandatory to have a version, any version, in newer Rails.

File: spec/support/rails_app/Gemfile (hunk 2 above)
Comment (altintx) [reply spec/support/rails_app/Gemfile:28]: Had no output about deprecations and no impact on unit tests. 3.20 is when Arm64 native support appeared in PB so it'll become increasingly important as more folks upgrade in the future.

**Review by lorgio-jimenez — APPROVED** [review]
looks great!

**Review by notnotdrew — APPROVED** [review]

**Pull request merged**